File: books/c-how-to-program/ch18/Composition/Employee.cpp
// Fig. 18.11: Employee.cpp
// Employee class member-function definitions.
#include <iostream>
#include "Employee.h"
using namespace std;

Employee::Employee(const string &first, const string &last,
    const Date &dateOfBirth, const Date &dateOfHire)
    : firstName(first),
      lastName(last),
      birthDate(dateOfBirth),
      hireDate(dateOfHire)
{
    cout << "Employee object constructor: "
        << firstName << ' ' << lastName << endl;
}

void Employee::print() const
{
    cout << lastName << ", " << firstName << "  Hired: ";
    hireDate.print();
    cout << "  Birthday: ";
    birthDate.print();
    cout << endl;
}

Employee::~Employee()
{
    cout << "Employee object destructor: "
        << lastName << ", " << firstName << endl;
}

File: books/java-how-to-program/src/ch08/examples/composition/Employee.java
// Fig. 8.8: Employee.java
// Employee class with references to other objects.
package ch08.examples.composition;

public class Employee {
    private String firstName;
    private String lastName;
    private Date birthDate;
    private Date hireDate;

    public Employee(String firstName, String lastName,
        Date birthDate, Date hireDate) {
        
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.hireDate = hireDate;
    }

    @Override
    public String toString() {
        return String.format("%s, %s Hired: %s BirthDay: %s", 
            lastName, firstName, hireDate, birthDate);
    }
    
}